src/main.ts

```diff
@@ -102,26 +103,17 @@ async function main() {
 function startUpPrompt(wsUrl: string, chainName: string, implName: string) {
 	const { logger } = Log;
 	const { config } = SidecarConfig;
+
 	/**
-	 * Best effort list of known public nodes that do not encourage high traffic
-	 * sidecar installations connecting to them for non - testing / development purposes.
+	 * Retrieving public endpoints from @polkadot/apps-config/endpoints
 	 */
-	const publicWsUrls: string[] = [
-		'wss://rpc.polkadot.io',
-		'wss://cc1-1.polkadot.network',
-		'wss://kusama-rpc.polkadot.io',
-		'wss://cc3-5.kusama.network',
-		'wss://fullnode.centrifuge.io',
-		'wss://crab.darwinia.network',
-		'wss://mainnet-node.dock.io',
-		'wss://mainnet1.edgewa.re',
-		'wss://rpc.kulupu.corepaper.org/ws',
-		'wss://main1.nodleprotocol.io',
-		'wss://rpc.plasmnet.io/',
-		'wss://mainnet-rpc.stafi.io',
-		'wss://rpc.subsocial.network',
-		'wss://full-nodes-lb.kilt.io:443',
-	];
+	const endpoints = createWsEndpoints(<T = string>(): T => '' as unknown as T);
+	const publicWsUrls: string[] = endpoints
+		.map(function (endpoint) {
```

Comment:
Hey @niklasad1! 
Thank you for your review! I replaced the `while` loop with `map` to iterate through `endpoints`.